### PR TITLE
tests: update interface tests to remove extra checks and normalize tests

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes:
* Remove all the checks that are done after a connection and disconnection (this is already checked as part of the snap_connect and snap_disconnect tests).
* install_local function used to pack and install local snaps
* Remove not needed echo
* Change the check of interfaces default behaviour for the  autoconnection to support the scenario when there are other snaps also connected to the interface (that happens when testing caracalla gadget)
* Add new lines between task tags

### CODE REVIEW

**Comment by plars** [comment]
tests/main/snap-connect has a similar issue to interfaces-cli that I mentioned earlier. Something like this should work I think (with an opportunistic fix for the test that was supposed to try connecting withtout specifying the slot):
diff --git a/tests/main/snap-connect/task.yaml b/tests/main/snap-connect/task.yaml
index 6aadc0d71..398a70bdb 100644
--- a/tests/main/snap-connect/task.yaml
+++ b/tests/main/snap-connect/task.yaml
@@ -11,17 +11,17 @@ prepare: |
     fi

 execute: |
-    CONNECTED_PATTERN=':home +home-consumer'
+    CONNECTED_PATTERN='(?s):home.+home-consumer'

     echo "The plug can be connected to a matching slot of OS snap without snap:slot argument"
     snap connect home-consumer:home
-    snap interfaces | MATCH "$CONNECTED_PATTERN"
+    snap interfaces -i home| grep -Pzq "$CONNECTED_PATTERN"

     snap disconnect home-consumer:home

     echo "The plug can be connected to a matching slot with slot name omitted"
-    snap connect home-consumer:home
-    snap interfaces | MATCH "$CONNECTED_PATTERN"
+    snap connect home-consumer
+    snap interfaces -i home| grep -Pzq "$CONNECTED_PATTERN"

     snap disconnect home-consumer:home
     snap tasks --last=disconnect| MATCH "Disconnect .* from core:home"
@@ -29,7 +29,7 @@ execute: |
     echo "The plug can be connected to a slot on the core snap using abbreviated syntax"
     snap connect home-consumer:home :home

-    snap interfaces | MATCH "$CONNECTED_PATTERN"
+    snap interfaces -i home| grep -Pzq "$CONNECTED_PATTERN"

     snap tasks --last=connect| MATCH "Connect home-consumer:home to core:home"

File: tests/main/interfaces-framebuffer/task.yaml [hunk 1]
```diff
@@ -11,20 +11,16 @@ prepare: |
     snap try $TESTSLIB/snaps/test-snapd-framebuffer
 
 execute: |
-    CONNECTED_PATTERN=":framebuffer +test-snapd-framebuffer"
-    DISCONNECTED_PATTERN="\- +test-snapd-framebuffer:framebuffer"
+    echo "The plug is not connected by default"
+    snap interfaces | MATCH "\- +test-snapd-framebuffer:framebuffer"
```
Comment (zyga) [inline tests/main/interfaces-framebuffer/task.yaml]: Seeing this over and over I start to agree with @plars that using `snap interfaces -I framebuffer` here and everywhere else would just make this less error prone.

Comment (sergiocazzolato) [reply tests/main/interfaces-framebuffer/task.yaml]: @zyga, I agree it is more robust, I'll implemet that in a new PR due to is it a big change.

Thanks for reviewing

Comment (bboozzoo) [reply tests/main/interfaces-framebuffer/task.yaml]: Makes sense to do it an follow-up. This one is already big.

**Pull request merged**